examples/vqe2d_lattice.py

```diff
@@ -0,0 +1,81 @@
+import time
+import optax
+import tensorcircuit as tc
+from tensorcircuit.templates.lattice import SquareLattice, get_compatible_layers
+from tensorcircuit.templates.hamiltonians import heisenberg_hamiltonian
+
+# Use JAX for high-performance, especially on GPU.
+K = tc.set_backend("jax")
+tc.set_dtype("complex64")
+# On Windows, cotengra's multiprocessing can cause issues.
+tc.set_contractor("cotengra-8192-8192", parallel=False)
+
+
+def run_vqe():
+    n, m, nlayers = 4, 4, 6
+    lattice = SquareLattice(size=(n, m), pbc=True, precompute_neighbors=1)
+    h = heisenberg_hamiltonian(lattice, j_coupling=[1.0, 1.0, 0.8])  # Jx, Jy, Jz
+    nn_bonds = lattice.get_neighbor_pairs(k=1, unique=True)
+    gate_layers = get_compatible_layers(nn_bonds)
+
+    def singlet_init(circuit):
+        # A good initial state for Heisenberg ground state search
+        nq = circuit._nqubits
+        for i in range(0, nq - 1, 2):
+            j = (i + 1) % nq
+            circuit.X(i)
+            circuit.H(i)
+            circuit.cnot(i, j)
+            circuit.X(j)
+        return circuit
+
+    def vqe_forward(param):
+        """
+        Defines the VQE ansatz and computes the energy expectation.
+        The ansatz consists of nlayers of RZZ, RXX, and RYY entangling layers.
+        """
+        c = tc.Circuit(n * m)
+        c = singlet_init(c)
+
+        for i in range(nlayers):
+            for layer in gate_layers:
+                for j, k in layer:
+                    c.rzz(int(j), int(k), theta=param[i, 0])
```

Comment:
we should find a better way to make each parameter different for each gate, a naive and also bruteforce way is to use a one-dimensional tensor for `param`, and for each use of `param`, we call `param[count]` then `count+=1`